cli/azd/internal/cmd/deploy.go

```diff
@@ -288,14 +288,9 @@ func (da *DeployAction) Run(ctx context.Context) (*actions.ActionResult, error)
 					da.console.StopSpinner(ctx, stepMessage, input.StepFailed)
 					return err
 				}
-
-				// Append package artifacts
-				if err := serviceContext.Package.Add(packageResult.Artifacts...); err != nil {
-					return err
-				}
```

Comment:
The `ServiceManager` is now fully responsible for appending any newly generated artifacts from the service level operations.